app/src/main/java/appricottsoftware/clarity/fragments/SettingFragment.java

```diff
@@ -52,11 +58,68 @@ public void onViewCreated(View view, @Nullable Bundle savedInstanceState) {
         // Initialize view lookups, listener
         // TODO: write a listener for when fragment is done drawing UI
         tvProgress.setVisibility(View.GONE);
+        setUp();
+    }
+
+    private void setUp() {
         getPodcastLength();
         setSeekBar();
         getOldEmail();
         setEmailListener();
         setPasswordListener();
+        setDeleteAccountListener();
+    }
+
+    private void setDeleteAccountListener() {
+        final int uid = ClarityApp.getSession(getContext()).getUserID();
+        btDeleteAccount.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.i(TAG, "User clicked on Delete Account Button");
+                // Creating an
+                AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+                builder.setCancelable(true);
+                builder.setTitle("Delete Your Account?");
+                builder.setMessage("\nRemoving this account will PERMANENTLY DELETE all user's data from Clarity! \n\n" +
+                                    "Are you sure you want to PERMANENTLY delete your account?");
+
+                // If user presses Yes
+                builder.setPositiveButton("Confirm",
+                    new DialogInterface.OnClickListener() {
+                        @Override
+                        public void onClick(DialogInterface dialog, int which) {
+                            // TODO Implmenet on confirmation request
+                            ClarityApp.getRestClient().deleteAccount(uid, getContext(), new JsonHttpResponseHandler() {
+                                @Override
+                                public void onSuccess(int statusCode, Header[] headers, JSONObject response) {
+                                    Log.i(TAG, "User: " + uid + " was successfully deleted");
+                                    ClarityApp.getSession(getContext()).setUserID(-1);
+                                    Intent loginActivityIntent = new Intent(getContext(), LoginActivity.class);
+                                    startActivity(loginActivityIntent);
```

Comment:
Please call finish() after startActivity(). Calling finish() results in the HomeActivity being popped from the backstack, which will prevent the user from returning to HomeActivity after pressing the back button from LoginActivity. Only calling startActivity() pushes LoginActivity onto the backstack on top of HomeActivity.

Reply:
Done

Reply:
Done